app.js

```diff
@@ -0,0 +1,170 @@
+//---------------------------------------------------------------------
+//5. write a global function that: renders the data to the page for the participant to view at the conclusion of the survey.
+//6. write a prototype method that: counts the number of times an object was displayed.
+//7. write a prototype method that: counts the number of times an object was clicked.
+//8. write a prototype method that: calculates a percentage based on the number of times an object was displayed versus the number of times it was clicked.
+//9. control the flow of the survey by concluding it after the participant has made 25 selections (clicks).
+//-----------------------------------------------------------------------
+
+//GLOBAL VARIABLES
+var sectionEl = document.getElementById('survey-content');
+
+//CONSTRUCTOR FUNCTION for generating products
+function Product(productName, imageURL, elementID) {
+  this.productName = productName;
+  this.productImage = imageURL;
+  this.elementID = elementID;
+  this.figureEl;
+  this.imageEl;
+  this.numberOfDisplays = 0;
+  this.numberOfClicks = 0;
+  this.clicksVsDisplayPercentage = 0;
+}
+
+//PROPERTIES ARRAY: this is where I store my objects' properties
+var productProperties = [
+  ['bag', 'img/bag.jpg', 'bagEl'],
+  ['banana', 'img/banana.jpg', 'bananaEl'],
+  ['bathroom', 'img/bathroom.jpg', 'bathroomEl'],
+  ['boots', 'img/boots.jpg', 'bootsEl'],
+  ['breakfast', 'img/breakfast.jpg', 'breakfastEl'],
+  ['bubblegum', 'img/bubblegum.jpg', 'bubblegumEl'],
+  ['chair', 'img/chair.jpg', 'chairEl'],
+  ['cthulhu', 'img/cthulhu.jpg', 'cthulhuEl'],
+  ['dog duck', 'img/dog-duck', 'dogDuckEl'],
+  ['dragon', 'img/dragon.jpg', 'dragonEl'],
+  ['pen', 'img/pen.jpg', 'penEl'],
+  ['pet sweep', 'img/pet-sweep.jpg', 'petSweepEl'],
+  ['scissors', 'img/scissors.jpg', 'scissorsEl'],
+  ['shark', 'img/shark.jpg', 'sharkEl'],
+  ['sweep', 'img/sweep.png', 'sweepEl'],
+  ['tauntaun', 'img/tauntaun.jpg', 'tauntaunEl'],
+  ['unicorn', 'img/unicorn.jpg', 'unicornEl'],
+  ['usb', 'img/usb.gif', 'usbEl'],
+  ['water can', 'img/water-can.jpg', 'waterCanEl'],
+  ['wine glass', 'img/wine-glass.jpg', 'wineGlassEl']
+];
+
+//PRODUCTS ARRAY: new objects are declared inside of the array
+var products = [];
+
+//this array stores the random numbers for the currently displayed products
+var currentRandomNumbers = [];
+
+//this array stores the random numbers from the 3 previously displayed products
+var oldRandomNumbers = [];
+
+//GLOBAL FUNCTION that iterates with constructor function and pushes objects into PRODUCTS ARRAY
+function createProducts() {
+
+  for (var i = 0; i < productProperties.length; i++) {
+
+    var newProduct = new Product(productProperties[i][0], productProperties[i][1], productProperties[i][2]);
+
+    products.push(newProduct);
+  }
+};
+
+//generates the random number used to access products from products array
+function getRandomNumber() {
+  var randomNumber = Math.floor(Math.random() * productProperties.length);
```

Comment:
with just one line of logic, you don't even need to assign this as a variable.  you could just write `return Math.floor(Math.random() * productProperties.length);`